### PR TITLE
docs(storage): add W&B artifact & provenance reference

### PR DESCRIPTION
**Pull request by ktinubu**
## Summary

Writes `docs/reference/artifact-provenance-reference.md` — the companion the storage-provenance spec §13 promised ("TBD (#122)") but never shipped. Now writable since the W&B artifact/lineage code has landed (#1471, #1472, #1473, #1474, #128).

The reference documents the **landed code patterns**:

- **Artifact catalog** — `dataset` / `model` / `eval-results` names, builders, and R2 references.
- **Reference pattern** — why R2 objects attach as `s3://` with `checksum=False`.
- **Metadata convention** (spec §6) — what lives in `artifact.metadata` per type.
- **Lineage DAG** — `log_artifact` producers + `use_artifact` consumers, with opt-in consumed edges.
- **`${wandb:…}` resolver** — resolving a model artifact to a cached local checkpoint for resume.
- **Code map** — every symbol → file.

Also resolves the dangling §13 reference link in the spec and registers the doc in `doc-map.yaml` against its five source modules so `doc-drift` tracks it.

## Validation

- Docs-only change — exempt from the tdd/code-health/simplify chain per AGENTS.md.
- `make format` clean (mdformat, prettier, codespell pass).
- Every factual claim fact-checked against the implementation; all symbols, names, metadata keys, and behaviors match.

Refs #1565

### CODE REVIEW

**Comment by ktinubu** [comment]
## Verification Results

Pure-docs PR — there is no runtime surface to exercise. The end-to-end **promise of a reference doc is cross-reference integrity**: a reader following it finds the cited code, the spec link resolves, and every `doc-map` source pattern (what `doc-drift` consumes) matches a real file. `mkdocs build --strict` is N/A — `reference/` is in `exclude_docs` in `mkdocs.yml`, so the new doc is intentionally not in the site build yet (same as its sibling reference docs).

All checks run via `uv run python /tmp/verify_doc.py` (full script below).

- [x] **Every symbol/file the doc cites resolves in the codebase** *(Level 3 — query live codebase; this is the reference doc's core promise. A failure would mean the doc points a reader at code that doesn't exist.)*
  ```
  A. symbol/file claims checked: 14; failures: 0
  ```
  14 `(symbol → file)` pairs asserted: each file exists and `def <symbol>` is defined in it (`build_dataset_artifact`, `build_model_artifact`, `build_eval_results_artifact`, `_consumed_artifact_refs`, `use_input_artifacts`, `_resolve_wandb_checkpoint`, `register_resolvers`, `_cache_key`, `pin_wandb_run_id`, `log_wandb_provenance`, …).

- [x] **Spec §13 reference link resolves to the new doc** *(Level 4 — resolve the relative link and stat the target; a typo would fail.)*
  ```
  B. spec §13 link target: ../reference/artifact-provenance-reference.md -> exists=True
  ```

- [x] **`doc-map.yaml` is valid YAML and every new source pattern matches ≥1 real file** *(Level 2 — parsed with `yaml.safe_load` and globbed each pattern; this is the exact mapping `doc-drift` reads to detect code→doc drift.)*
  ```
  C. doc-map sources checked: 5
  ALL CHECKS PASS
  ```
  All 5 source patterns (`finalize_dataset.py`, `train.py`, `eval.py`, `logging_utils.py`, `utils.py`) resolve.

<details><summary>verification script (<code>/tmp/verify_doc.py</code>)</summary>

```python
# (symbol -> file) pairs the doc promises a reader will find; spec §13 link
# resolution; doc-map source-pattern globbing. Exits non-zero on any failure.
import pathlib, re, sys, glob, yaml
# ... see PR branch; asserts `def <symbol>` in each cited file, resolves the
# spec relative link, and globs each doc-map source pattern.
```
</details>

**Facts also independently fact-checked** against the implementation during the pre-PR review — all names, metadata keys, and behaviors matched (0 mismatches).

File: docs/reference/artifact-provenance-reference.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+## 1. Artifact Catalog
+
+| Type           | Name pattern               | Built by                                         | R2 reference                             |
+| -------------- | -------------------------- | ------------------------------------------------ | ---------------------------------------- |
+| `dataset`      | `data-{dataset_config_id}` | `build_dataset_artifact` (`finalize_dataset.py`) | split `.h5` / shard prefix + `stats.npz` |
+| `model`        | `model-{train_config_id}`  | `build_model_artifact` (`train.py`)              | checkpoint prefix (opt-in, see §3)       |
+| `eval-results` | `eval-{eval_config_id}`    | `build_eval_results_artifact` (`eval.py`)        | output-dir prefix                        |
+
+The `{*_config_id}` is the config filename stem, resolved via `resolve_run_config_id(cfg)` for train/eval and `spec.task_name` for datasets. The `{*_wandb_run_id}` lives in `artifact.metadata`, never in the artifact name — W&B auto-versions (`:v0`, `:v1`, …) so re-running the same config yields the next version.
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply docs/reference/artifact-provenance-reference.md]: Good catch — fixed in 548bbe06. The builders don't copy the run id into `artifact.metadata`, so I reworded: the artifact name carries the config id, the producing run's id is pinned via `pin_wandb_run_id`, and I noted that spec §4 reserves the metadata slot but it isn't wired yet.

File: docs/reference/artifact-provenance-reference.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+use_input_artifacts(loggers, _consumed_artifact_refs(cfg))  # e.g. [("data-diva-v1", "latest")]
+```
+
+Consumed edges are opt-in: training reads `consumed_dataset_config_id` (null by default → no edge), so a run without it set records no input lineage and never calls `use_artifact`.
+
+______________________________________________________________________
+
+## 5. Resolving an Artifact to a Checkpoint
+
+The `${wandb:<ref>}` OmegaConf resolver (`utils/utils.py`, registered in `register_resolvers`) turns a model-artifact ref into a local checkpoint path — used for `ckpt_path=wandb:model-flow-simple:best` resume:
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply docs/reference/artifact-provenance-reference.md]: Fixed in 548bbe06. Reworded the prose to require the `${wandb:…}` interpolation and note that the bare `wandb:…` form is passed through literally and never resolved, matching the YAML example below it.

File: docs/reference/artifact-provenance-reference.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+______________________________________________________________________
+
+## 6. Aliases (spec §4)
+
+| Alias         | Set by           | When                                  |
+| ------------- | ---------------- | ------------------------------------- |
+| `:latest`     | W&B (automatic)  | every `log_artifact` call             |
+| `:best`       | training script  | when the val metric improves          |
+| `:production` | promote workflow | when a model is promoted to a Release |
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply docs/reference/artifact-provenance-reference.md]: Fixed in 564f7cd2. You're right — the landed code applies only `:latest` (W&B-automatic); `_log_model_artifact` logs with no `aliases=[…]` and the promote workflow that would set `:production` isn't implemented (#1566). Added a **Status** column marking `:best`/`:production` as planned (spec-reserved) vs the one landed alias.

File: docs/reference/artifact-provenance-reference.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+## 3. Metadata Convention (spec §6)
+
+`artifact.metadata` holds properties of the artifact itself — never run hyperparameters (those go in `wandb.config`) or final metrics (`wandb.summary`).
+
+| Artifact       | Metadata keys                                                |
+| -------------- | ------------------------------------------------------------ |
+| `dataset`      | `shard_count`, `n_samples`, `git_sha`                        |
+| `model`        | `git_sha`                                                    |
+| `eval-results` | scalar summary metrics (`_eval_summary_metrics`) + `git_sha` |
+
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply docs/reference/artifact-provenance-reference.md]: Fixed the metrics-convention contradiction in 472431e3: §3 now says final metrics live in `wandb.summary`, with `eval-results` documented as the exception that also copies a scalar summary into `artifact.metadata` (via `_eval_summary_metrics`) for filtering.

On the `||` point: the tables actually use a *single* leading pipe — `cat -A` shows each row starts with one `| `, and mdformat (pre-commit) validates them, so they render correctly on GitHub. No change needed there.

File: docs/reference/artifact-provenance-reference.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+______________________________________________________________________
+
+## 1. Artifact Catalog
+
+| Type           | Name pattern               | Built by                                         | R2 reference                             |
+| -------------- | -------------------------- | ------------------------------------------------ | ---------------------------------------- |
+| `dataset`      | `data-{dataset_config_id}` | `build_dataset_artifact` (`finalize_dataset.py`) | split `.h5` / shard prefix + `stats.npz` |
+| `model`        | `model-{train_config_id}`  | `build_model_artifact` (`train.py`)              | checkpoint prefix (opt-in, see §3)       |
+| `eval-results` | `eval-{eval_config_id}`    | `build_eval_results_artifact` (`eval.py`)        | output-dir prefix                        |
+
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply docs/reference/artifact-provenance-reference.md:27]: This is a false positive — the Artifact Catalog table uses a single leading pipe, not `||`. `cat -A` confirms each row begins with one `| `, and the mdformat pre-commit hook (which rejects malformed tables) passes, so the table renders as the intended 4 columns on GitHub. No change needed.

File: docs/reference/artifact-provenance-reference.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+Consumed edges are opt-in: training reads `consumed_dataset_config_id` (null by default → no edge), so a run without it set records no input lineage and never calls `use_artifact`.
+
+______________________________________________________________________
+
+## 5. Resolving an Artifact to a Checkpoint
+
+The `${wandb:<ref>}` OmegaConf resolver (`utils/utils.py`, registered in `register_resolvers`) turns a model-artifact ref into a local checkpoint path. To resume, point `ckpt_path` at a `${wandb:…}` interpolation — the bare `wandb:…` form is passed through literally and never resolved:
+
+```yaml
+ckpt_path: ${wandb:model-flow-simple:best}
```
[the remark this answers is not in the document]
Comment (ktinubu) [reply docs/reference/artifact-provenance-reference.md]: Fixed in 08ebab25 — changed the resume example to `${wandb:model-flow-simple:latest}`, which resolves against the only alias the landed code actually sets (W&B-automatic `:latest`), consistent with §6.

File: docs/reference/artifact-provenance-reference.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (ktinubu) [reply docs/reference/artifact-provenance-reference.md:27]: Confirmed false positive: `grep -n '||' docs/reference/artifact-provenance-reference.md` returns zero matches and `cat -A` shows every row starts with a single `| `. mdformat (pre-commit) validates the tables, so they render with the intended columns — no leading empty column exists to fix.
